### PR TITLE
Multi select activities

### PR DESCRIPTION
**Pull request by dogcomplex**
# Overview

This PR includes the following proposed change(s):

- https://github.com/bcgov/invasivesbc/issues/317

I so far have not opted to follow the "Should always be able to see which records I've selected (i.e. should float to the top)" directive from the ticket - as that's a bit of a pain, and I'm not especially convinced it's a good user experience.  For now just including a count of the number selected in the button, and I'd suggest maybe putting that selected list on the bulk edits page itself in another PR if that's a big concern still.  Can add that in a follow-up PR though, whichever UX choice we make.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Locally

## Screenshots

2021-01-19 12_44_27-InvasivesBC - Search

Unsupported records have no checkbox (disabled and hidden) while maintaining a margin:
2021-01-19 13_19_58-InvasivesBC - Search

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have made corresponding changes to the documentation
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/src/components/activities-list/SearchActivitiesList.tsx [hunk 1]
```diff
@@ -85,12 +91,28 @@ const SearchActivitiesList: React.FC<ISearchActivitiesList> = (props) => {
   return (
     <List>
       {props.activities.map((activity) => {
+        const isChecked = editIds.includes(activity._id);
+        // Temporarily limit bulk editing:
+        const bulkEditIsDisabled = !(activity.activitySubtype === ActivitySubtype.Observation_PlantTerrestrial);
```
Comment (sdevalapurkar) [inline app/src/components/activities-list/SearchActivitiesList.tsx]: why are we checking only for plant terrestrial here? not too sure I understand this logic

File: app/src/components/activities-list/SearchActivitiesList.tsx [hunk 2]
```diff
@@ -85,12 +91,28 @@ const SearchActivitiesList: React.FC<ISearchActivitiesList> = (props) => {
...
+        console.log(activity);
```
Comment (sdevalapurkar) [inline app/src/components/activities-list/SearchActivitiesList.tsx]: lets remove this line
```suggestion
```

File: app/src/components/search/SearchResultsList.tsx [hunk 3]
```diff
@@ -10,12 +11,38 @@ interface ISearchResultsList {
 }
 
 const SearchResultsList: React.FC<ISearchResultsList> = (props) => {
+  const [editIds, setEditIds] = useState<any[]>([]);
+  const history = useHistory();
+
+  const bulkEditActivities = (editIds: any) => {
```
Comment (sdevalapurkar) [inline app/src/components/search/SearchResultsList.tsx]: let's rename this to something like
`navigateToBulkEditPage` or something along those lines so it doesn't seem like this function is doing the bulk editing itself

File: app/src/components/search/SearchResultsList.tsx [hunk 4]
```diff
@@ -10,12 +11,38 @@ interface ISearchResultsList {
...
     <Box>
-      <Typography variant="h5">
-        {(!props.totalItems && 'No Results Found') || `${props.totalItems} Matching Results Found`}
-      </Typography>
-      <SearchActivitiesList activities={props.results} />
+      <Box
+        style={{
+          justifyContent: 'space-between',
+          display: 'flex',
+          alignItems: 'flex-end'
```
Comment (sdevalapurkar) [inline app/src/components/search/SearchResultsList.tsx]: can we put this in a class?

File: app/src/components/search/SearchResultsList.tsx [hunk 5]
```diff
@@ -10,12 +11,38 @@ interface ISearchResultsList {
...
+        }}>
+        <Typography variant="h5">
+          {(!props.totalItems && 'No Results Found') || `${props.totalItems} Matching Results Found`}
```
Comment (sdevalapurkar) [inline app/src/components/search/SearchResultsList.tsx]: why not just:
```
`${props.totalItems}` Matching Results Found
```

**Comment by sdevalapurkar** [comment]
also I see these which should be fixed prior to merge: https://sonarcloud.io/project/issues?id=bcgov_invasivesbc&pullRequest=455&resolved=false&types=CODE_SMELL

File: app/src/components/search/SearchResultsList.tsx (hunk 5 above)
Comment (dogcomplex) [reply app/src/components/search/SearchResultsList.tsx]: 🤷  wasn't my doing

File: app/src/components/activities-list/SearchActivitiesList.tsx [hunk 6]
```diff
@@ -85,12 +91,35 @@ const SearchActivitiesList: React.FC<ISearchActivitiesList> = (props) => {
...
+
+        // Temporarily limit bulk editing:
+        const allowedBulkEditSubtypes = [
+          ActivitySubtype.Observation_PlantTerrestrial,
+          ActivitySubtype.Treatment_ChemicalPlant,
+          ActivitySubtype.Treatment_MechanicalPlant,
+          ActivitySubtype.Treatment_BiologicalPlant,
+          ActivitySubtype.Treatment_BiologicalDispersalPlant
+        ];
+        const bulkEditIsDisabled = allowedBulkEditSubtypes.indexOf(activity.activitySubtype) === -1;
```
Comment (sdevalapurkar) [inline app/src/components/activities-list/SearchActivitiesList.tsx]: why not
```
!allowedBulkEditSubtypes.includes(activity.activitySubtype)
```

**Comment by sdevalapurkar** [comment]
@dogcomplex what do you think of adding a column for the checkbox instead of adding it to the existing icon column?

**Comment by dogcomplex** [comment]
> @dogcomplex what do you think of adding a column for the checkbox instead of adding it to the existing icon column?

Sure? whatever, really.  Could change the icon to have a little checkmark in the corner when it's selected too, as another idea.  I'm assuming the current look is likely to be scrapped in the longterm, so wouldn't spend too much time perfecting it right now.  If I'm going to that effort, I'd consolidate the whole thing with the IAPP tables and make a universal grid table component too.  So - not gonna bother stuffing it into this PR.

**Review by sdevalapurkar — APPROVED** [review]

**Pull request merged**